test/gtest/deepbench_lstm.cpp

```diff
@@ -0,0 +1,193 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <utility>
+
+#include "lstm.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest_common.hpp>
+#include <gtest/gtest.h>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_DEEPBENCH)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_FLOAT)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_HALF)
+
+namespace deepbench_lstm {
+using EnvType = std::tuple<std::pair<miopen::env::MIOPEN_TEST_DEEPBENCH, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_FLOAT, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_HALF, std::string_view>>;
+
+static bool Skip(miopenDataType_t prec)
+{
+    bool flag = !miopen::IsEnabled(ENV(MIOPEN_TEST_DEEPBENCH));
+    switch(prec)
+    {
+    case miopenFloat: return flag && !miopen::IsEnabled(ENV(MIOPEN_TEST_FLOAT));
+    case miopenHalf: return flag && !miopen::IsEnabled(ENV(MIOPEN_TEST_HALF));
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+    default: MIOPEN_THROW("Unsupported datatype");
+    }
+    return true;
+}
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+void SetEnv(EnvType env_vars)
+{
+    std::apply(
+        [](const auto&... pairs) {
+            (..., (miopen::UpdateEnvVar(pairs.first, std::string_view(pairs.second))));
+        },
+        env_vars);
+}
+
+auto GetTestCases(std::string precision)
+{
+    const auto env = std::tuple{
+        std::pair{ENV(MIOPEN_TEST_DEEPBENCH), std::string_view("ON")},
+        std::pair{ENV(MIOPEN_TEST_FLOAT),
+                  precision == "--float" ? std::string_view("ON") : std::string_view("OFF")},
```

Comment:
How this should work? What is the use case?
I can't understand this code design, why do you need an environment variables?

Reply:
@shurale-nkn See #1843. CTest is passing env vars to gtests:
https://github.com/ROCm/MIOpen/blob/25a399f9127a67567a027c900dc1da1c6564f61d/test/gtest/CMakeLists.txt#L33

Reply:
@shurale-nkn But in this specific case this code is incorrect.

@cderb The original test does not **_set_** the environment so this one should do the same. What we need to do is **_check_** the environment. I'll cover this in my review.

Reply:
@atamazov current implementation will rewrite any passed variable.

Reply:
@shurale-nkn Yeah it should be fixed

Reply:
There appears to have been some changes while I was away, I'll be making some corrections shortly.